File: leetcode/1051_height_checker.cpp
// Copyright (c) 2019 Jacopo Notarstefano

#include <algorithm>
#include <cassert>
#include <vector>

using namespace std;

class Solution {
 public:
    int heightChecker(vector<int>& heights) {
        vector<int> sortedHeights(heights.begin(), heights.end());
        sort(sortedHeights.begin(), sortedHeights.end());

        int result = 0;

        for (int i = 0; i < heights.size(); i++) {
            if (heights[i] != sortedHeights[i]) {
                result++;
            }
        }

        return result;
    }
};

int main() {
    auto solution = Solution();

    vector<int> heights = {1, 1, 4, 2, 1, 3};

    assert(3 == solution.heightChecker(heights));
}

File: leetcode/1051_height_checker.py
# -*- coding: utf-8 -*-


class Solution:
    def heightChecker(self, heights):
        return sum(x != y for x, y in zip(heights, sorted(heights)))


if __name__ == "__main__":
    solution = Solution()

    assert 3 == solution.heightChecker([1, 1, 4, 2, 1, 3])
